libp2p/crypto/common.go

```diff
@@ -0,0 +1,54 @@
+package crypto
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/secp256k1"
+	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// ConvertPrivateKeyToLibp2pPrivateKey will convert common private key to libp2p private key
+func ConvertPrivateKeyToLibp2pPrivateKey(privateKey crypto.PrivateKey) (libp2pCrypto.PrivKey, error) {
+	if check.IfNil(privateKey) {
+		return nil, ErrNilPrivateKey
+	}
+
+	p2pPrivateKeyBytes, err := privateKey.ToByteArray()
+	if err != nil {
+		return nil, err
+	}
+
+	return libp2pCrypto.UnmarshalSecp256k1PrivateKey(p2pPrivateKeyBytes)
+}
+
+// CreateRandomP2PIdentity creates a random p2p crypto components
+func CreateRandomP2PIdentity() ([]byte, core.PeerID, error) {
```

Comment:
we do not need this. What we need is a converter between the provided private key and the peer ID

Reply:
it was kept only for testing, removed it